src/ripple/app/main/Application.cpp

```diff
@@ -1609,6 +1610,55 @@ bool ApplicationImp::setup()
         }
     }
 
+    if (shardStore_)
+    {
+        using namespace boost::filesystem;
+
+        auto stateDb(
+            RPC::ShardArchiveHandler::getDownloadDirectory(*config_)
+            / stateDBName);
+
+        try
+        {
+            if (exists(stateDb) &&
+                is_regular_file(stateDb) &&
+                !RPC::ShardArchiveHandler::hasInstance())
+            {
+                auto handler = RPC::ShardArchiveHandler::getInstance(
+                    *this,
+                    *m_jobQueue);
```

Comment:
With my initial submission it actually was created and destroyed automatically based on the initiation and completion of downloads. With feedback I had to make it a stoppable which means no more destruction until application end.

Reply:
If I make the SAH a member of App, then would need synchronization around the construction of the thing. Is it ok to just add a new mutex to the AppImp for that purpose?

Reply:
Deferring for now

Reply:
There's already a static `instance_mutex_` in SAH, so I don't see any harm in moving it to AppImp. (There may be an argument for reusing `AppImp::mut_` instead, since that's only used to determine when to stop running and isn't likely to conflict, but that's not a big decision either way.)